integration/cluster_test.go

```diff
@@ -416,3 +419,77 @@ func clusterMustProgress(t *testing.T, membs []*member) {
 		mcancel()
 	}
 }
+
+func TestTransferLeader(t *testing.T) {
```

Comment:
The tests are overly complicated i feel.

We probably only need one simple test to start with:

1. create a 3 members cluster
2. kill leader
3. wait on the stop chan of the killed leader
4. issue put to one of the other member
5. put should succeed within one election timeout

without this patch, put cannot finish within one election timeout.

/cc @heyitsanthony sounds good?